### PR TITLE
Handle backslashes correctly in no-hex-escape and escape-case

### LINKED ISSUE #85: escape-case and no-hex-escape false positives – need to handle backslashes
package.json:

```json
{
  "private": true,
  "eslintConfig": {
    "env": {
      "es6": true
    },
    "parserOptions": {
      "ecmaVersion": 2017,
      "sourceType": "module"
    },
    "plugins": [
      "unicorn"
    ],
    "rules": {
      "unicorn/escape-case": "error",
      "unicorn/no-hex-escape": "error"
    }
  },
  "scripts": {
    "test": "eslint test.js"
  },
  "devDependencies": {
    "eslint": "^3.15.0",
    "eslint-plugin-unicorn": "^2.0.1"
  }
}
```

test.js:

```js
"\\ufeff"
"\\\\xa0"
```

Output:

```
$ npm test

> @ test /home/lydell/bugs/eslint-plugin-unicorn-bugs
> eslint test.js


/home/lydell/bugs/eslint-plugin-unicorn-bugs/test.js
  1:1  error  Use uppercase characters for the value of the escape sequence  unicorn/escape-case
  2:1  error  Use uppercase characters for the value of the escape sequence  unicorn/escape-case
  2:1  error  Use unicode escapes instead of hexadecimal escapes             unicorn/no-hex-escape

✖ 3 problems (3 errors, 0 warnings)

npm ERR! Test failed.  See above for more details.
```

Expected: No errors.

### PR DESCRIPTION
**Pull request by nkovacs**
Fixes #85

### CODE REVIEW

**Comment by nkovacs** [comment]
This conflicts with #109, the merge of both branches is here: https://github.com/nkovacs/eslint-plugin-unicorn/commit/d118afcd0c37a71c0fd0f6ee3f8871e01581533a

**Comment by sindresorhus** [comment]
@nkovacs Can you look into the failing test?

**Comment by nkovacs** [comment]
Fixed.

File: rules/no-hex-escape.js [hunk 1]
```diff
@@ -2,7 +2,7 @@
 'use strict';
 
 function checkEscape(context, node, value) {
-	const fixedValue = typeof value === 'string' ? value.replace(/\\x/g, '\\u00') : value;
+	const fixedValue = typeof value === 'string' ? value.replace(/((^|[^\\])(\\\\)*)\\x/g, '$1\\u00') : value;
```
Comment (SamVerschueren) [inline rules/no-hex-escape.js]: @sindresorhus If we use non-capturing groups (`?:`), does it impact performance?

Comment (nkovacs) [inline rules/no-hex-escape.js]: Non-capturing should be faster: https://developer.mozilla.org/en/docs/Web/JavaScript/Reference/Global_Objects/RegExp

**Comment by sindresorhus** [comment]
Thank you @nkovacs :)

**Pull request merged**